lg_mirror/webapps/playback/index.html

```diff
@@ -28,18 +30,40 @@
 
 var streamDiv = document.getElementById('stream');
 
+var rosbridgeHost = getParameterByName('rosbridge_host', String, 'localhost');
+var rosbridgePort = getParameterByName('rosbridge_port', Number, 9090);
+var rosbridgeSecure = getParameterByName('rosbridge_secure', stringToBoolean, false);
+var rosbridgeUrl = getRosbridgeUrl(rosbridgeHost, rosbridgePort, rosbridgeSecure);
+var rosInstanceName = getParameterByName('ros_instance_name', String);
 var webVideoServerUri = getParameterByName('web_video_server', String, 'http://localhost:8080');
 var viewport = getParameterByName('played_back_viewport', String);
 if (!viewport) {
   console.error('Must specify viewport param');
 }
 
+var ros = new ROSLIB.Ros();
+ros.connect(rosbridgeUrl);
+
+var readinessTopic = new ROSLIB.Topic({
+  ros: ros,
+  name: '/director/window/ready',
+  messageType: 'std_msgs/String'
+});
+readinessTopic.advertise();
+var readinessMsg = new ROSLIB.Message({
+   data: rosInstanceName
+});
+
 // Must match the form of lg_mirror.utils.get_viewport_base_topic()
 // ros_compressed stream type will append /compressed
 var streamTopic = '/lg_mirror/viewport/' + viewport;
 var streamUrl = webVideoServerUri + '/stream?type=ros_compressed&topic=' + streamTopic;
 
 var img = document.createElement('img');
+img.onload = function(ev) {
```

Comment:
to properly emit a signal that's independent from ondomContentLoaded, you need to send message to the extension - it's documented [here](https://github.com/EndPointCorp/lg_ros_nodes/tree/master/lg_common/src/lg_common/extensions/ros_window_ready#application-generated-message)